src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -578,7 +578,40 @@ def _pem_password_cb(self, password):
         userdata = _PasswordUserdata(password=password)
         return _pem_password_cb, userdata
 
-    def _mgf1_hash_supported(self, algorithm):
+    def _get_oaep_supported_padding_algorithms(backend):
+        # Verified test Vectors for OAEP only cover SHA-1 and SHA-2 currently
+        # with OpenSSL >= 1.0.2.
+        if backend._lib.OpenSSL_version_num() >= 0x10002001:
+            _oaep_supported_padding_algorithms = (
+                hashes.SHA1,
+                hashes.SHA224,
+                hashes.SHA256,
+                hashes.SHA384,
+                hashes.SHA512
+            )
+        else:
+            _oaep_supported_padding_algorithms = ( hashes.SHA1, )
+
+        return _oaep_supported_padding_algorithms
+
+    def _oaep_mgf1_hash_supported(self, algorithm):
+        if self._lib.Cryptography_HAS_MGF1_MD:
```

Comment:
@reaperhulk It's not clear to me what the implications of this check are, _oaep_hash_supported will return it supports SHA-1 and SHA-2 in the current code but _oaep_mgf1_hash_supported will only return it supports SHA-1 when this HAS_MGF1_MD is false, is that the right behaviour?

Is HAS_MGF1_MD linked tightly to the OpenSSL version? And if so is it actually covered by _get_oaep_supported_padding_algorithms check for version >= 0x10002001 and so this HAS_MGF1_MD isn't needed and can be removed etc?